homeassistant/components/bmw_connected_drive/coordinator.py

```diff
@@ -31,6 +31,8 @@ class BMWDataUpdateCoordinator(DataUpdateCoordinator[None]):
     """Class to manage fetching BMW data."""
 
     account: MyBMWAccount
+    read_only: bool
+    _entry: ConfigEntry
```

Comment:
We already have `self.config_entry` so I think we should remove this and use the one `DataUpdateCoordinator` already provides

Reply:
The `super()` call below should also include this.

Reply:
Ah yes, I remember reading this one but didn't realize the impact.

However I have to type it, as mypy otherwise complains:
```
homeassistant/components/bmw_connected_drive/coordinator.py:93: error: Item "None" of "ConfigEntry[Any] | None" has no attribute "data"  [union-attr]
homeassistant/components/bmw_connected_drive/coordinator.py:98: error: Argument 1 to "async_update_entry" of "ConfigEntries" has incompatible type "ConfigEntry[Any] | None"; expected "ConfigEntry[Any]"  [arg-type]
```

If you'd prefer some other handling (like an if-statement in `_update_config_entry_refresh_token()` please let me know.

Reply:
That's done in e4f1ca2